library/src/main/java/com/karumi/headerrecyclerview/HeaderRecyclerViewAdapter.java

```diff
@@ -59,10 +59,26 @@
   }
 
   /**
+   * If you don't need header feature, you can bypass overriding this method.
+   */
+  protected VH onCreateHeaderViewHolder(ViewGroup parent, int viewType) {
+    return null;
+  }
+
+  protected abstract VH onCreateItemViewHolder(ViewGroup parent, int viewType);
+
+  /**
+   * If you don't need footer feature, you can bypass overriding this method.
+   */
+  protected VH onCreateFooterViewHolder(ViewGroup parent, int viewType) {
+    return null;
+  }
+
+  /**
    * Invokes onBindHeaderViewHolder, onBindItemViewHolder or onBindFooterViewHOlder methods based
    * on the position param.
    */
-  @Override public void onBindViewHolder(VH holder, int position) {
+  @Override public final void onBindViewHolder(VH holder, int position) {
```

Comment:
Neither this one. Declaring this method as final, as I said before, we are restricting the component usage based on inheritance. Removing this final modifier we can also remove the usage of powermock :)